main.go

```diff
@@ -1,55 +1,168 @@
 package main
 
 import (
+	"io"
 	"os"
+	"strconv"
 
-	"github.com/segmentio/go-source"
-	"github.com/segmentio/kit"
-	"github.com/segmentio/kit/config"
-	"github.com/segmentio/kit/schema"
-	"golang.org/x/net/context"
-	"gopkg.in/mgo.v2"
+	"github.com/Sirupsen/logrus"
+	"github.com/asaskevich/govalidator"
+	"github.com/segmentio/objects-go"
+	"github.com/tj/docopt"
+	"github.com/tj/go-sync/semaphore"
 )
 
-var Version = "0.1.0"
+const (
+	Version = "0.0.1-beta"
+)
+
+var usage = `
+Usage:
+  mongodb
+    [--debug]
+    [--init]
+    [--concurrency=<c>]
+    [--schema=<schema-path>]
+    --write-key=<segment-write-key>
+    --hostname=<hostname>
+    --port=<port>
+    --username=<username>
+    --password=<password>
+    --database=<database>
+  mongodb -h | --help
+  mongodb --version
+
+Options:
+    "github.com/segmentio/source-db-lib/internal/domain"
+  -h --help                   Show this screen
+  --version                   Show version
+  --write-key=<key>           Segment source write key
+  --concurrency=<c>           Number of concurrent table scans [default: 1]
+  --hostname=<hostname>       Database instance hostname
+  --port=<port>               Database instance port number
+  --username=<username>       Database instance username
+  --password=<password>       Database instance password
+  --database=<database>       Database instance name
+  --schema=<schema-path>	  The path to the schema json file [default: schema.json]
+
+`
 
 func main() {
-	kit.Run(schema.Service{
-		Name:    "mongodb",
-		Version: Version,
-		Config: schema.Config{
-			{
-				Key:      "url",
-				Required: true,
-			},
-			{
-				Key:      "database",
-				Required: true,
-			},
-		},
-		Handler: run,
-	})
-}
+	app := &MongoDB{};
+	defer app.Close()
+
+	m, err := docopt.Parse(usage, nil, true, Version, false)
+	if err != nil {
+		logrus.Error(err)
+		return
+	}
 
-func run() {
-	url := config.Get("url").(string)
-	database := config.Get("database").(string)
+	segmentClient := objects.New(m["--write-key"].(string))
+	defer segmentClient.Close()
 
-	session, err := mgo.Dial(url)
-	check(err)
+	setWrapperFunc := func(o *objects.Object) {
```

Comment:
we should probably also panic out or at least log at the error level in this case, right?